lib/Map/TableStructure.js

```diff
@@ -279,33 +287,37 @@ TableStructure.prototype.toArrayOfColumns = function() {
 * Return data as an array of rows of formatted data, eg. [ ['x', 'y'], ['1', '12,345'], ['2.1', '20'] ].
 * @param {String} [dateFormatString] If present, override the standard date format with a string (see https://www.npmjs.com/package/dateformat)
 *                 Eg. "isoDateTime" or "dd mmm yyyy HH:MM:ss"
+* @param {Integer[]} [rowNumbers] An array of row numbers to return. Defaults to all rows.
 * @returns {Object} An array of rows of formatted data, the first of which is the column names.
 */
-TableStructure.prototype.toArrayOfRows = function(dateFormatString) {
+TableStructure.prototype.toArrayOfRows = function(dateFormatString, rowNumbers) {
     if (this.columns.length < 1) {
         return;
     }
     var that = this;
     var rows = that.columns[0].values.map(function(value0, rowIndex) {
-        return that.columns.map(function(column) {
-            if (dateFormatString && column.type === VarType.TIME) {
-                return dateFormat(column.dates[rowIndex], dateFormatString);
-            }
-            return column._formattedValues[rowIndex];
-        });
-    });
+        if (!defined(rowNumbers) || rowNumbers.indexOf(rowIndex) >= 0) {
```

Comment:
Once again this might not affect performance that much, but wouldn't it make more sense to loop over the rowNumbers and get that row out of `that.columns[0].values` for each one, instead of looping over `that.columns[0].values` and for each one checking all of `rowNumbers` to see if that row is in it?